### PR TITLE
Support for multi-line headers and titles

### PR DESCRIPTION
**Pull request by GiudGiud**
- Changed log.cpp for multi-line titles and headers, using existing function create_multi_line_msg
- Added arguments for "filling the paddings" on both sides of the title and header messages (that are centered) (removed)
- took into account parity of number of characters to make title well centered and nice even when odd/even number of characters in it (or in level) (removed)
- updated test_plotter.py and results_true.dat

To meet Issue #229 raised by @samuelshaner

### CODE REVIEW

**Comment by GiudGiud** [comment]
Is there any way to check what Travis sees? I m having a hard time guessing

**Comment by GiudGiud** [comment]
@tjlaboss when you have time

I suspect the training PR I did a year ago was just too messy for the others to accept it. 
This version gives up on having multi-line centered and with the nice paddings on both sides, but at least it addresses the bug.

**Comment by tjlaboss** [comment]
In your comment dated [Jun 16, 2017](https://github.com/mit-crpg/OpenMOC/pull/331#issuecomment-309190233), I assume that's referring to `travis-ci` and not me?

I'm not really qualified to review the C++ code, but I can try to diagnose why tests are failing with Python2 and passing with Python3.

**Comment by GiudGiud** [comment]
Ok I ll ask Wenbin for the C++ once I find a way to tag him!

With regards to the test, it's odd, the tests pass on my machine on both python 3.6 and 2.7. I ll look at the test suite tomorrow

**Comment by tjlaboss** [comment]
@GiudGiud , I might have found the problem. [All of the tests that failed](https://travis-ci.org/mit-crpg/OpenMOC/jobs/382888002#L1044) were plotting using some form of colormap. As of Matplotlib 2.0.0, [the 'spectral' color map will raise an error](https://matplotlib.org/api/api_changes.html#color-maps). I ran into this issue when I updated my Python libraries last week. See PR #339.

Counterintuitively, it could be that the libraries Travis CI is using Python 2.7 have been updated more recently than those for Python 3.6.

**Comment by tjlaboss** [comment]
> Is there any way to check what Travis sees?

Yes: 

image

**Comment by GiudGiud** [comment]
Ha yes I had seen that one. 
I wanted a job log with more information than which test has failed.

Ok let's get PR #339 through first then

**Comment by tjlaboss** [comment]
Closing and re-opening to make Travis do his thing

**Comment by GiudGiud** [comment]
Good idea. I had already corrected some typos in material.cpp for the same purpose.

File: src/log.cpp [hunk 1]
```diff
@@ -378,28 +378,52 @@ void log_printf(logLevel level, const char* format, ...) {
       }
     case (HEADER):
       {
-        int size = strlen(buffer);
-        int halfpad = (line_length - 4 - size) / 2;
-        std::string pad1 = std::string(halfpad, header_char);
-        std::string pad2 = std::string(halfpad +
-                           (line_length - 4 - size) % 2, header_char);
         std::string prefix = std::string("[  HEADER ]  ");
```
Comment (wuwenbin2006) [inline src/log.cpp]: Use level_prefix for consistency

**Comment by GiudGiud** [comment]
Resetting PR to re-run the tests. 
@wuwenbin2006 anything else you would like changed ? If not, please merge

**Pull request merged**